app/client/src/workers/Evaluation/evaluate.ts

```diff
@@ -75,11 +72,25 @@ const topLevelWorkerAPIs = Object.keys(self).reduce((acc, key: string) => {
 
 function resetWorkerGlobalScope() {
   for (const key of Object.keys(self)) {
-    if (topLevelWorkerAPIs[key]) continue;
-    if (key === "evaluationVersion") continue;
-    if (extraLibraries.find((lib) => lib.accessor === key)) continue;
-    // @ts-expect-error: Types are not available
-    delete self[key];
+    if (topLevelWorkerAPIs[key] || DOM_APIS[key]) continue;
+    //TODO: Remove this once we have a better way to handle this
+    if (
+      key === "evaluationVersion" ||
+      key === "window" ||
+      key === "document" ||
+      key === "location" ||
+      key === "localStorage"
+    )
```

Comment:
Suggestion:- we could reduce the condition by writing the same code as below. 
```suggestion
    if (
      ["evaluationVersion", "window", "document", "location", "localStorage"].includes(key)
    )
```

Reply:
Done, removed localStorage from the list since we are not adding support for it in this iteration